src/main/java/uk/nhs/digital/nhsconnect/nhais/service/RegistrationConsumerService.java

```diff
@@ -21,21 +22,41 @@
 
     public void handleRegistration(MeshMessage meshMessage) {
         LOGGER.debug("Received Registration message: {}", meshMessage);
-        Interchange interchange = new EdifactParser().parse(meshMessage.getContent());
+        Interchange interchange;
+        try {
+            interchange = new EdifactParser().parse(meshMessage.getContent());
+        } catch (ToEdifactParsingException ex) {
+            createNackRecep(ex); //TODO implementation of that
+            return;
+        }
         LOGGER.debug("Parsed registration message into interchange: {}", interchange);
 
         var inboundState = InboundState.fromInterchange(interchange);
         if (!saveState(inboundState)) {
             return;
         }
 
+        sendSequenceNumbersToInboundSequenceManager(inboundState); //TODO implementation of that
+
         // recep producer service
         Parameters outputParameters = new EdifactToFhirService().convertToFhir(interchange);
         LOGGER.debug("Converted registration message into FHIR: {}", outputParameters);
         inboundGpSystemService.publishToSupplierQueue(outputParameters, inboundState.getOperationId());
         LOGGER.debug("Published inbound registration message to gp supplier queue");
     }
 
+    private void createNackRecep(ToEdifactParsingException ex) {
+        //this exception should be passed to RECEP producer to create NACK RECEP
```

Comment:
Logging and throwing an exception (so that the message is not acknowledged on the inbound mesh queue) is the correct behaviour. NACK RECEPs are not supported.

Reply:
removed